### PR TITLE
Support start with other IDEs in Github

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Support start with other IDEs, please update this description if you think it's necessary 🙏 

Relates https://github.com/gitpod-io/gitpod/issues/6706

TODO:

~~‼️ Auto detect recommended IDE options for user (need UX design too)~~ This one need Application of Gitpod and ask users for permission, not in this PR
- [x] ‼️ Do not use `#referrer:`, create another URL? or in the future? https://github.com/gitpod-io/gitpod/issues/9434
- [x] Support Gitlab
- [ ] Support Bitbucket
- [ ] Update UX
- [ ] Update description of IDEs

|Github|GitLab|
|:-:|:-:|
|image|image|

## Related Issue(s)

Fixes #

## How to test

**With NoVNC**

Open gitpod with this PR, see preview

**With local test**

1. Open gitpod with this PR
2. Download `gitpod.xpi` in project root, and unzip it
3. Open *Chrome Extensions Manage*
4. Install via *Load unpacked* with unzipped folder

## Release Notes

```release-note
Support start workspace with other IDEs in Github
```

### CODE REVIEW

**Comment by mustard-mh** [comment]
cc @gtsiolis

**Comment by mustard-mh** [comment]
I think this is a good onboarding(more than onboard) option https://github.com/gitpod-io/gitpod/issues/6707

cc @loujaybee

**Comment by loujaybee** [comment]
@mustard-mh this looks super interesting. We'd need to make sure we've got appropriate tracking in place. 

> I think this is a good onboarding(more than onboard) option https://github.com/gitpod-io/gitpod/issues/6707

It likely won't make as high of an impact, as not all users use the browser extension. That said, it does look like a nice way to elevate the other IDE's in the product experience 😀

**Comment by mustard-mh** [comment]
In my opinion, it is better to provide (recommended) this browser extension to users than to display the onboarding option.

It would be a bad experience if we let user select the onboarding option and then open another workspace with this option for them. Unless we do `Choose(change) IDEs after workspace started` first, so that users can just change IDEs. 

@loujaybee

File: src/injectors/github-injector.ts [hunk 1]
```diff
@@ -111,67 +116,97 @@ abstract class ButtonInjectorBase implements ButtonInjector {
...
+    genOptionsString(ideOptions: IDEOptions, host: string, originUrl: string) {
+        if (!ideOptions.clients) {
+            return []
+        }
+        return Object.entries(ideOptions.clients ?? {}).reduce((prev, [key, value]) => {
+            if (key === "vscode-insiders" || !value.desktopIDEs) {
+                return prev
+            }
+            prev.push(...value.desktopIDEs?.map(e => {
+                const url = `${host}/#referrer:${key}:${e}/${originUrl}`
```
[the remark this answers is not in the document]
Comment (akosyakov) [reply src/injectors/github-injector.ts:135]: thanks, please make sure to avoid using it for testing on gitpod.io too, it will disturb our data 🙏

**Comment by akosyakov** [comment]
Do we close it for now?

**Comment by gtsiolis** [comment]
> Do we close it for now?

@akosyakov I'd think so, as it sounds preferable to make the workspace start page more flexible instead and provide more user control and options there, see also https://github.com/gitpod-io/gitpod/issues/9534#issuecomment-1110102471 and [relevant discussion](https://gitpod.slack.com/archives/C01KGM9BH54/p1650970351290919) (internal).

🍊 🍊 🍊 🍊 

Thanks @mustard-mh for giving this a shot! It was great to also see this PR in action following design guidelines and patterns on each platform (GitHub and GitLab) for extending the dropdown options. 🧡

**Comment by akosyakov** [comment]
I also meant temporarily. If we don't plan to work or land something within a week. I think it is safe to close and reopen when it is important.

**Pull request closed without merging**